### PR TITLE
add encryption to private graph

### LINKED ISSUE #3: [Graph SDK] Add encryption Support
## Why
We need to be able to encrypt and decrypt private graph related data before storing it on chain.

### Acceptance criteria
- encryption data when Writing private graph
- decryption when reading private graph

#### more details
Details of encryption algorithm can be found [here](https://github.com/LibertyDSNP/spec/pull/240/files?short_path=0d85a9f#diff-0d85a9f3c7a6627c8d6f8fb14494b0c357781b77c4e3027b5078683a6ee87c75)

Current chosen crate is `crypto_box`

### PR DESCRIPTION
**Pull request by aramikm**
# Goal
The goal of this PR is  to add encryption and decryption to private graph
closes #3 
# Acceptence Criterea
- [x] encrypting when Writing private graph
- [x] decrypting when reading private graph
- [x] Tests added

### CODE REVIEW

File: core/src/dsnp/reader_writer.rs [hunk 1]
```diff
@@ -12,28 +13,40 @@ use anyhow::Result;
...
+	) -> Result<Vec<u8>>;
 }
 
 /// A utility to read/write data from and to Frequency chain specific implementation of DSNP
 pub struct Frequency;
 
+/// implementing DsnpReader for Frequency
 impl DsnpReader for Frequency {
+	/// using SealBox for encryption
+	type Encryption = SealBox;
```
Comment (aramikm) [inline core/src/dsnp/reader_writer.rs:48]: Is there a way to to assign the EncryptionBehavior implementation on struct level so we don't have to assign it one DsnpReader and DsnpWriter separately!?

**Review by wesbiggs — APPROVED** [review]
Looks good to me and nicely structured. I assume the distinction between privateFollows and privateConnections occurs at a level above this code? (The schemas/chunks should be the same.)

**Comment by aramikm** [comment]
@wesbiggs If I remember correctly we decided to distinguish between different graphs (connection types) with different schemas (Even if schema structure might be the same).

@wilwade Please let me know if this is still the case.

**Comment by aramikm** [comment]
Merging to keep the ball rolling! I'll address any additional comment is coming PRs.

**Pull request merged**